src/sas/qtgui/Calculators/GenericScatteringCalculator.py

```diff
@@ -48,7 +48,20 @@ def __init__(self, parent=None):
         self.sld_reader = sas_gen.SLDReader()
         self.pdb_reader = sas_gen.PDBReader()
         self.reader = None
-        self.sld_data = None
+        #sld data for nuclear and magnetic cases
+        self.nuc_sld_data = None
+        self.mag_sld_data = None
+        #verification information to avoid recalculating
+        #verification carried out whenever files are selected/deselected
+        #verification reset whenever a new files loaded
+        self.verification_occurred = False # has verification happened on these files
+        self.verified = False # was the verification successsful
+        # verification error label
+        #prevent layout shifting when widget hidden (could this be placed i nthe .ui file?)
```

Comment:
I've added the TODO on this in 3523a453409b6e1dea221f4ed5e476c638cd270f and 3f959d75efcc67543d1e354ee98aea4f97be425c. Normally when a widget is set as invisible, it is taken by the GUI to have zero size, this can cause the whole layout to shift very slightly. e.g. for the combobox:

![combobox_shift_behaviour](https://user-images.githubusercontent.com/57537928/126763444-f6d6dded-d5fc-4e57-9697-517ac957bc65.png)

This size policy makes the widget keep its size when invisible. To be honest the effect is very small, it just caught my eye when it did it the first few times. This fix wasn't actually preventing shifting in the case of the error label as it was adjusting to being two lines high, at least on my screen. In commit 3f959d75efcc67543d1e354ee98aea4f97be425c I have set two lines high to be the default size so that this shifting should not occur at all.

Reply:
It's a subtle shift in position, but I see now what it's doing.